internal/characters/ganyu/cons.go

```diff
@@ -13,31 +13,30 @@ const (
 	c1Key = "ganyu-c1"
 	c4Key = "ganyu-c4"
 	c6Key = "ganyu-c6"
+	c1ICD = "ganyu-c1-energy-icd"
 )
 
-func (c *char) c1() {
-	c.Core.Events.Subscribe(event.OnEnemyDamage, func(args ...interface{}) bool {
-		atk := args[1].(*combat.AttackEvent)
-		e, ok := args[0].(*enemy.Enemy)
-		if !ok {
-			return false
+//Ganyu C1: Taking DMG from a Charge Level 2 Frostflake Arrow or Frostflake Arrow Bloom decreases opponents' Cryo RES by 15% for 6s.
+//A hit regenerates 2 Energy for Ganyu. This effect can only occur once per Charge Level 2 Frostflake Arrow, regardless if Frostflake Arrow itself or its Bloom hit the target.
+func (c *char) c1() combat.AttackCBFunc {
```

Comment:
There's no need to introduce a custom icd for this if you use the same callback instance for both arrows. So in aimed.go just define something like `c1cb := c.c1()` there.
Also missing constellation check:
```
if c.Base.Cons < 1 {
    return nil 
}
done := false 
return func(a combat.AttackCB) {
    e:= a.Target.(*enemy.Enemy)
    if e.Type() != combat.TargettableEnemy {
        return
    }
    if done {
        return
    }
    done = true
    ...
}
```